catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -0,0 +1,171 @@
+                .build();
+    }
+
+    @Path("/{registryId}/publication/{sourceId}")
+    @DELETE
+    public Response unpublish(@PathParam("registryId") String registryId,
+            @PathParam("sourceId") String destinationId) {
+        if (StringUtils.isBlank(registryId) || StringUtils.isBlank(destinationId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry and destination ID cannot be blank")
+                    .build();
+        }
+
+        try {
+            registryPublicationService.unpublish(registryId, destinationId);
+        } catch (FederationAdminException e) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Error publishing node to destination " + destinationId)
+                    .build();
+        }
+
+        return Response.status(Response.Status.OK)
+                .build();
+    }
+
+    @Path("/{registryId}/report")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewRegInfoHtml(@PathParam("registryId") final String registryId,
+            @QueryParam("sourceId") final List<String> sourceIds) {
+        String html = "";
+        if (StringUtils.isBlank(registryId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry Id cannot be blank")
+                    .build();
+        }
+
+        RegistryPackageType registryPackage;
+        try {
+            registryPackage = federationAdminService.getRegistryObjectByRegistryId(registryId,
+                    sourceIds);
+        } catch (FederationAdminException e) {
+            String message = "Error getting registry package.";
+            LOGGER.error("{} For metacard id: '{}', optional sources: {}",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        if (registryPackage == null) {
+            String message = "No registry package was found.";
+            LOGGER.error("{} For metacard id: '{}', optional source ids: {}.",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(message)
+                    .build();
+        }
+
+        Map<String, Object> registryMap = reportHelper.buildRegistryMap(registryPackage);
+
+        try {
+            Handlebars handlebars = new Handlebars(templateLoader);
+            Template template = handlebars.compile("report");
+            html = template.apply(registryMap);
+        } catch (IOException e) {
+            LOGGER.error("Error compiling and applying report template.");
+        }
+
+        return Response.ok(html)
+                .build();
+    }
+
+    public void setRegistryPublicationService(
```

Comment:
No specific reason. Just personal preference. Haven't read anything concrete that details when to use argument injection vs property injection.